test/python/transpiler/test_high_level_synthesis.py

```diff
@@ -0,0 +1,275 @@
+            self.assertNotIn("op_a", ops.keys())
+            self.assertEqual(ops["id"], 2)
+            self.assertIn("op_b", ops.keys())
+            self.assertEqual(ops["op_b"], 1)
+
+    def test_default_config(self):
+        """Check the default behavior of HighLevelSynthesis, with
+        the default HighLevelSynthesisConfig specified. The behavior should
+        be the same as without config."""
+        qc = self.create_circ()
+        mock_plugin_manager = MockPluginManager
+        with unittest.mock.patch(
+            "qiskit.transpiler.passes.synthesis.high_level_synthesis.HighLevelSynthesisPluginManager",
+            wraps=mock_plugin_manager,
+        ):
+            hls_config = HLSConfig()
+            pm = PassManager([HighLevelSynthesis(hls_config=hls_config)])
+            tqc = pm.run(qc)
+            ops = tqc.count_ops()
+            # OpA's default method replaces by "id", OpB has no default method
+            self.assertNotIn("op_a", ops.keys())
+            self.assertEqual(ops["id"], 2)
+            self.assertIn("op_b", ops.keys())
+            self.assertEqual(ops["op_b"], 1)
+
+    def test_non_default_config(self):
+        """Check the default behavior of HighLevelSynthesis, specifying
+        non-default synthesis methods for OpA and for OpB.
+        """
+        qc = self.create_circ()
+        mock_plugin_manager = MockPluginManager
+        with unittest.mock.patch(
+            "qiskit.transpiler.passes.synthesis.high_level_synthesis.HighLevelSynthesisPluginManager",
+            wraps=mock_plugin_manager,
+        ):
+            hls_config = HLSConfig(op_a=[("repeat", {"n": 2})], op_b=[("simple", {})])
+            pm = PassManager([HighLevelSynthesis(hls_config=hls_config)])
+            tqc = pm.run(qc)
+            ops = tqc.count_ops()
+            self.assertNotIn("op_a", ops.keys())
+            self.assertNotIn("op_b", ops.keys())
+            self.assertEqual(ops["id"], 4)
+            self.assertEqual(ops["cx"], 1)
+
+    def test_synthesis_returns_none(self):
+        """Check that when synthesis method is specified but returns None,
+        the operation does not get synthesized.
+        """
+        qc = self.create_circ()
+        mock_plugin_manager = MockPluginManager
+        with unittest.mock.patch(
+            "qiskit.transpiler.passes.synthesis.high_level_synthesis.HighLevelSynthesisPluginManager",
+            wraps=mock_plugin_manager,
+        ):
+            hls_config = HLSConfig(op_a=[("repeat", {})])
+            pm = PassManager([HighLevelSynthesis(hls_config=hls_config)])
+            tqc = pm.run(qc)
+            ops = tqc.count_ops()
+            # The repeat method for OpA without "n" specified returns None.
+            self.assertIn("op_a", ops.keys())
+            self.assertIn("op_b", ops.keys())
+
+    def test_use_default_on_unspecified_is_false(self):
+        """Check that when use_default_on_unspecified is False, the default synthesis
```

Comment:
Is there a corresponding test for when `use_default_on_unspecified` is `True`?

Reply:
Thanks, I have added this test as well.